src/C++/FieldConvertors.h

```diff
@@ -577,16 +607,26 @@ struct UtcTimeOnlyConvertor
           throw FieldConvertError();
     }
 
-    return std::string( result, showMilliseconds ? sizeof( result ) : 8 );
+    if( showMicroseconds )
+    {
+      if( integer_to_string_padded ( result + 13, 3, millis )
+          != result + 13 )
+          throw FieldConvertError();
+    }
+
+    size_t length = showMicroseconds ? sizeof( result ) : showMicroseconds ? 12 : 8;
```

Comment:
You have a copy/paste mistake here.
size_t length = showMicroseconds ? sizeof( result ) : **showMilliseconds**? 12 : 8;